controllers/machineselector_controller.go

```diff
@@ -218,7 +218,9 @@ func (r *MachineInventorySelectorReconciler) findAndAdoptInventory(ctx context.C
 		return nil
 	}
 
-	patchBase := client.MergeFrom(mInventory.DeepCopy())
+	// Ensure we patch the latest version otherwise we could erratically
+	// set the ownership of an already owned inventory
+	patchBase := client.MergeFromWithOptions(mInventory.DeepCopy(), client.MergeFromWithOptimisticLock{})
```

Comment:
@fgiudici after the chat I realize this is also needed, was a meaningful chat :+1: I have not seen any race on this resource within the CI, but it doesn't mean it could not happen, this prevents two selectors picking the same inventory.